src/logger/log_entry.py

```diff
@@ -0,0 +1,166 @@
+from datetime import datetime
+
+import marshmallow as marsh
+from peewee import CharField, DateTimeField, ForeignKeyField, IntegerField
+
+from src.base_model import BaseModel
+from src.implementor.models import Run
+
+
+class LogEntry(BaseModel):
+    """Represents a single log entry. Used to log messages from the simulation."""
+
+    class Schema(BaseModel.Schema):
+        """The marshmallow schema for the LogEntry model."""
+
+        timestamp = marsh.fields.DateTime(required=True)
+        message = marsh.fields.String(required=True)
+        run_id = marsh.fields.UUID(required=True)
+
+        def _make(self, data: dict) -> "LogEntry":
+            return LogEntry(**data)
+
+    timestamp = DateTimeField(null=False, default=datetime.now())
+
+    message = CharField(null=False)
+    run_id = ForeignKeyField(Run, null=False)
+
+
+class TrainSpawnLogEntry(LogEntry):
+    """A LogEntry that represents the spawning of a train."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainSpawnLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainSpawnLogEntry":
+            return TrainSpawnLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+
+
+class TrainRemoveLogEntry(LogEntry):
+    """A LogEntry that represents the removal of a train."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainRemoveLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainRemoveLogEntry":
+            return TrainRemoveLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+
+
+class TrainArrivalLogEntry(LogEntry):
+    """A LogEntry that represents the arrival of a train at a station."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainArrivalLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+        station_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainArrivalLogEntry":
+            return TrainArrivalLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+    station_id = IntegerField(null=False)
```

Comment:
Please discuss with @SaturnHafen or @CR1337 if the id of a platform/station is an integer, uuid or string 😇 I'm unsure at the moment.
(Please update it in the other classes too 🥳 )

Reply:
As these are not generated by yaramo, we should probably stick with human-readable identifiers (string-names)

Reply:
Will also change it to a string :)